### PR TITLE
New right panel visual language

### PR DESCRIPTION
**Pull request by germain-gg**
Aligning the right panel with the new visual language. This also updates the old room header to align it with the base card header.

The "search" triggers the old in `RoomView` search and is a deliberate compromise made with @nadonomy . 

This is not expected to go behind a labs flag and will be rolled out straight away as all the functionality are meant to work the same, we are only adapting the visuals and improving contrast in a lot of places.

Screenshot 2023-10-03 at 14 26 33
Screenshot 2023-10-03 at 14 26 14

## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
Here's what your changelog entry will look like:

## ✨ Features
 * New right panel visual language ([\#11664](https://github.com/matrix-org/matrix-react-sdk/pull/11664)).

### CODE REVIEW

**Review by nadonomy — COMMENTED** [review]
Adding as a PR comment for now, so @americanrefugee @janogarcia can consider as part of their more formal review and we can funnel into relevant iteration/issues:

- Room header cursor: Doesn't switch from pointer to hand reliably. Need to hover over the room title I think, instead of just anywhere in the header.
- Searching for messages: 
    - Faux search bar triggers some kind of browser autocomplete for me after clicking. Should be labelled 'Search for messages or similar'. 
    - Initially suggested to re-stylise as a button (hover states etc) similar to the faux search button above the room list. But, that's not a standard component. 
    - From internal discussion, advocate to just lose the faux search bar altogether and rely on the button row. We'll improve search later.
- Hairline stroke: The harline stroke in the people & files views makes the layout feel too grid-like. I think we should hide it as a part of this PR as it effects basically every sub view, including profiles. We can trivially add the stroke back when implementing the new tab designs properly.
- Invite button: The invite button looks too visually strong imo. Can we tone it down to a text button or perhaps a fully rounded primary button?

Otherwise, looking/feeling on the right track!

File: package.json [hunk 1]
```diff
@@ -52,6 +52,7 @@
         "test": "jest",
         "test:cypress": "cypress run",
         "test:cypress:open": "cypress open",
+        "test:screenshot:local": "PERCY_TOKEN=a04f2cc2bbfa5d01df2e234622896ad47bff8433a26c563ef551d2e1c15a171f PERCY_BRANCH=local yarn percy exec --parallel -- yarn test:cypress",
```
Comment (kerryarchibald) [inline package.json:55]: Should this be here?

File: src/components/views/right_panel/BaseCard.tsx [hunk 2]
```diff
@@ -86,11 +86,13 @@ const BaseCard: React.FC<IProps> = forwardRef<HTMLDivElement, IProps>(
         return (
             <CardContext.Provider value={{ isCard: true }}>
                 <div className={classNames("mx_BaseCard", className)} ref={ref} onKeyDown={onKeyDown}>
-                    <div className="mx_BaseCard_header">
-                        {backButton}
-                        {closeButton}
-                        {header}
-                    </div>
+                    {header !== null && (
```
Comment (kerryarchibald) [inline src/components/views/right_panel/BaseCard.tsx:89]: Are there cases where we want a BaseCard without back and close buttons?

**Review by kerryarchibald — APPROVED** [review]
Left some questions, but looks good!

**Comment by americanrefugee** [comment]
@nadonomy asked "Invite button: The invite button looks too visually strong imo. Can we tone it down to a text button or perhaps a fully rounded primary button?"
- @janogarcia and I decided to change the secondary button outline to Light/color/border/interactive/Secondary, as this is a more subtle treatment but still is still accessible

**Comment by americanrefugee** [comment]
For Search...

- Clicking on the search bar in the right panel should ONLY activate and focus the existing search banner below the header, and not allow you to enter text in both places
- The text should not be medium/semi-bold, but Web/font/body/md/Regular

**Comment by americanrefugee** [comment]
@germain-gg I just discussed this further with @janogarcia 

Scrap what I wrote previously and do this instead...

- If the user has the new right panel enabled, but NOT the room header:
   - Remove the search bar from the right panel, and keep the search icon in the old room header
- If the user has the new room header enabled:
   - Change the search field to a [medium quick action button](https://www.figma.com/file/rTaQE2nIUSLav4Tg3nozq7/Compound-Web-Components?type=design&node-id=1912-33459&mode=design&t=Rowy8l3Q4Xwa0LYC-4)

Here's an example:
Web - Search messages

**Comment by andybalaam** [comment]
Most recent failure looks like https://github.com/vector-im/element-web/issues/25485 but that issue is not clear what error happens, so I'll log a new one.

**Comment by andybalaam** [comment]
Flaky test issue: https://github.com/vector-im/element-web/issues/26407

**Comment by andybalaam** [comment]
The same test failed again. Re-kicking here: https://github.com/matrix-org/matrix-react-sdk/actions/runs/6587323232 - if it's consistently failing we need to figure out why, but if it's flaking we can force-merge.

**Pull request merged**